### PR TITLE
【BIT】转换能力补齐：paddle.nn.functional.conv[1 2 3]d

### PR DESCRIPTION
**Pull request by cangtianhuang**
### 1. [paddle.nn.functional.conv1d](https://www.paddlepaddle.org.cn/documentation/docs/zh/develop/api/paddle/nn/functional/conv1d_cn.html)

查询 [torch.nn.functional.conv1d 与 paddle.nn.functional.conv1d 对照表](https://www.paddlepaddle.org.cn/documentation/docs/zh/develop/guides/model_convert/convert_from_pytorch/api_difference/functional/torch.nn.functional.conv1d.html) ，发现是 `paddle 参数更多`，注册并编写 Conv1dRule 类

查阅文档发现，paddle 多支持 data_format 参数，使用 permute 调换输入与输出维度顺序

paddle 与 torch 的多个参数用法不同：stride、dilation 若为列表，需转为 tuple 类型

此外，padding 参数具有更加丰富的形式：当 padding 为 “SAME” 或 “VALID” 时，torch 也支持此设置，直接转写为小写；当 padding 为长度为 1 的列表时，转为 tuple 类型；当 padding 为长度为 2 的列表时，代表非对称填充，torch 对应的 api 不支持非对称填充，因此需使用 torch.nn.functional.pad 对 torch 的输入进行手动填充

### 2. [paddle.nn.functional.conv2d](https://www.paddlepaddle.org.cn/documentation/docs/zh/develop/api/paddle/nn/functional/conv2d_cn.html)

查询 [torch.nn.functional.conv2d 与 paddle.nn.functional.conv2d 对照表](https://www.paddlepaddle.org.cn/documentation/docs/zh/develop/guides/model_convert/convert_from_pytorch/api_difference/functional/torch.nn.functional.conv2d.html) ，发现是 `paddle 参数更多`，注册并编写 Conv3dRule 类

与 paddle.nn.functional.conv1d 类似，但是参数类型略有不同，维度更多，区别如下：

paddle 多支持 data_format 参数，使用 permute 调换输入与输出维度顺序

当 padding 为 “SAME” 或 “VALID” 时，直接转写为小写；当 padding 为长度为 2 的列表时，转为 tuple 类型；当 padding 为长度为 4 的列表时，使用 torch.nn.functional.pad 对 torch 的输入进行手动填充；当 padding 为长度为 4、元素为长度为 2 的嵌套列表时，代表对输入的所有维度的非对称填充，因此根据 data_format 提取填充大小，对 torch 的输入进行手动填充

### 3. [paddle.nn.functional.conv3d](https://www.paddlepaddle.org.cn/documentation/docs/zh/develop/api/paddle/nn/functional/conv3d_cn.html)

查询 [torch.nn.functional.conv3d 与 paddle.nn.functional.conv3d 对照表](https://www.paddlepaddle.org.cn/documentation/docs/zh/develop/guides/model_convert/convert_from_pytorch/api_difference/functional/torch.nn.functional.conv3d.html) ，发现是 `paddle 参数更多`，注册并编写 Conv3dRule 类

与 paddle.nn.functional.conv2d 类似，但是参数类型略有不同，维度更多，区别如下：

paddle 多支持 data_format 参数，使用 permute 调换输入与输出维度顺序

当 padding 为 “SAME” 或 “VALID” 时，直接转写为小写；当 padding 为长度为 3 的列表时，转为 tuple 类型；当 padding 为长度为 6 的列表时，使用 torch.nn.functional.pad 对 torch 的输入进行手动填充；当 padding 为长度为 5、元素为长度为 2 的嵌套列表时，根据 data_format 提取填充大小，对 torch 的输入进行手动填充

## 测试结果

paddle.nn.functional.conv[1 2 3]d 大部分配置通过 accuracy 测试，移入 support2torch_1.txt 中；数条配置有极小的精度问题，移入 accuracy_error_1.txt 中 

image

### CODE REVIEW

**Review by Cutelemon6 — APPROVED** [review]
LGTM

**Review by wanghuancoder — APPROVED** [review]
LGTM
